### PR TITLE
feat(document): add web document for preset / custom docs

### PR DESCRIPTION
**Pull request by xxs-wallace**
### Background
We want to add unique constraint for `name` field. But it is impossible to add unique constraint in exist table because the origin filed of this table has no value.

The next step, we will add some preset document for user to crawler and search.

The origin table `web_crawler_urls` will be replaced by new table `web_documents` after all api interface is ready.

### CODE REVIEW

File: ee/tabby-db/src/web_crawler.rs
Comment (wsxiaoys) [inline ee/tabby-db/src/web_crawler.rs:1]: Please create a new file `web_document.rs` for such changes.

File: ee/tabby-db/src/lib.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ mod user_completions;
 mod user_events;
 mod users;
 mod web_crawler;
+mod web_document;
```
Comment (wsxiaoys) [inline ee/tabby-db/src/lib.rs:43]:
```suggestion
mod web_documents;
```

File: ee/tabby-db/src/web_document.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    pub id: i64,
+    pub name: String,
+    pub url: String,
+    pub is_preset: bool,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+}
+
+impl DbConn {
+    pub async fn list_web_document(
```
Comment (wsxiaoys) [inline ee/tabby-db/src/web_document.rs:20]:
```suggestion
    pub async fn list_web_documents(
```

**Review by wsxiaoys — CHANGES_REQUESTED** [review]
Some minor nits, otherwise LGTM

File: ee/tabby-schema/src/schema/constants.rs [hunk 3]
```diff
@@ -5,7 +5,8 @@ lazy_static! {
     pub static ref REPOSITORY_NAME_REGEX: Regex = Regex::new("^[a-zA-Z][\\w.-]+$").unwrap();
     pub static ref USERNAME_REGEX: Regex =
         Regex::new(r"^[^0-9±!@£$%^&*_+§¡€#¢¶•ªº«\\/<>?:;|=.,]{2,20}$").unwrap();
-    pub static ref WEB_DOCUMENT_NAME_REGEX: Regex = Regex::new(r"^[A-Za-z][A-Za-z0-9\ ]*$").unwrap();
+    pub static ref WEB_DOCUMENT_NAME_REGEX: Regex =
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/constants.rs:8]: for a quick follow up, the space can only present individually, e.g `a b c` is ok, but `a b  c` is not allowed.

Comment (xxs-wallace) [reply ee/tabby-schema/src/schema/constants.rs:8]: Done

**Pull request merged**